src/ppom-controller.ts

```diff
@@ -470,6 +494,20 @@ export class PPOMController extends BaseControllerV2<
     this.update((draftState) => {
       draftState.chainStatus = chainStatus;
     });
+  }
+
+  /*
+   * The function adds new network to chainStatus list.
+   */
+  #onNetworkChange(networkControllerState: NetworkState): void {
+    this.#chainId = addHexPrefix(networkControllerState.providerConfig.chainId);
+    this.#updateChainStatus(this.#chainId);
+    Object.values(networkControllerState.networkConfigurations).forEach(
+      (networkConfig) => {
+        const chainId = addHexPrefix(networkConfig.chainId);
+        this.#updateChainStatus(chainId);
+      },
+    );
     this.#deleteOldChainIds();
     this.#checkScheduleFileDownloadForAllChains();
   }
```

Comment:
`this.#updateChainStatus` should be called for all chains recently seen, also I think should we called for chainId in `usePPOM`

Reply:
What does "recently seen" mean here?

Reply:
`updateChainStatus` in `usePPOM` added here https://github.com/MetaMask/ppom-validator/pull/85/commits/383b8603ca08cf281e32d2f629e564ac145f9a3d

Reply:
Also wondering if there is a callback in network controller now which can be used to add networks to chainStatus as soon as user connects to it.

Reply:
Not from the network controller. Connecting to a network really just means that the network is permissioned for a certain dapp(s). Maybe we could get this from the PermissionsController?

Reply:
Hey @jpuri I think we may need to talk through a better way to identify a chain status since as Jiexi said above, in the upcoming multichain API 

> Connecting to a network really just means that the network is permissioned for a certain dapp(s)

if chainStatus is used to determine whether an incoming request might come for that chain we can't rely on "connection" status since it is really just has a dapp been granted permission to interact with that chain. This is very different from how it works now where we just permission accounts and then the user can "connect" to whatever chain and the dapp automatically gets "connected" to that chain. Long run we need to be able to serve requests that come from any of the "permissioned" chains many of which may not actually be active. So we should think of some protocol for warming up a given chain status when a first RPC request comes in for that chain and pausing any updates when it has been inactive for a while.

Reply:
In `chainStatus` we cache data for a chain so that for repeated requests we have it in cache, it is useful to add network to `chainStatus` as soon as user start interacting with it so that we have the data cache ready to be used when transaction is submitted.

Is there a way for us to identify when a user opens a DAPP connected to any chainId ?

Reply:
I think we could do this. That would mean going from connections -> origin -> networkClientId -> chainId and passing that to PPOM. 

Can we tackle this as part of a separate PR?

Reply:
> Can we tackle this as part of a separate PR?
➕